controllers/operator/routes.go

```diff
@@ -331,7 +331,10 @@ func (r *AuthenticationReconciler) reconcileRoute(authCR *operatorv1alpha1.Authe
 
 func (r *AuthenticationReconciler) ensureRouteDoesNotExist(ctx context.Context, authCR *operatorv1alpha1.Authentication, fields *reconcileRouteFields) (result *ctrl.Result, err error) {
 	reqLogger := logf.FromContext(ctx)
-	if fields.Name != IMCrtAuthRouteName {
+	if fields.Name == IMCrtAuthRouteName {
+		//do not delete certAuth route
+		return subreconciler.ContinueReconciling()
+	} else {
```

Comment:
You don't need this `else` because we're only dealing with the non-mTLS Routes for the rest of the function.

Reply:
removed else block